beacon-chain/db/kv/kv.go

```diff
@@ -147,6 +147,13 @@ func (s *Store) ClearDB() error {
 // Close closes the underlying BoltDB database.
 func (s *Store) Close() error {
 	prometheus.Unregister(createBoltCollector(s.db))
+
+	// Before DB closes, we should dump the cached state summary objects to DB.
```

Comment:
review

Reply:
We should add a context to this. Let's refactor the function signature of NewKVStore